### ISSUE TITLE
script.module.slyguy errors: NO_RESPONSE_ERROR and urllib3.util.ssl_ has no attribute _ssl_wrap_socket_impl

### ISSUE THREAD

**Issue by jpcandioti** [matthuisman/slyguy.addons#465]
Hi Matt,

Thanks for your work, my family use these addons a lot!

We live in Argentina and we use Disney+, Star+, and Paramount+ addons. Since 0.65.2 update of script.mocule.slyguy all addons doesn't work. I don't know what is wrong, but the error throw [here](https://github.com/matthuisman/slyguy.addons/blob/9fb34dd63b6086cbcf29342d2620d41963c48d3a/script.module.slyguy/resources/modules/slyguy/session.py#L397). Then, since 0.65.6 update of script.mocule.slyguy the error throw [here](https://github.com/matthuisman/slyguy.addons/blob/489d3281bb324f5fc33c0aa459b5473158abae9c/script.module.slyguy/resources/modules/slyguy/session.py#L40) "AttributeError: module 'urllib3.util.ssl_' has no attribute '__ssl_wrap_socket_impl'".

I back to version 0.65.1 and all they work again!

I have Ubuntu 20.04.6 LTS (GNU/Linux 5.4.0-150-generic x86_64) with Kodi Media Center 20.1 (20.1.0) Git:20230312-289ec664e3,

I hope my report is useful for you.

Regards.

**Comment by matthuisman** [comment]
0.65.6 was to fix the urllib error. You sure you've tried it? Please update to 0.65.6 and get me a debug log if still error

**Comment by jpcandioti** [comment]
Yes, this is the error message with 0.65.6.
```
2023-06-03 21:16:12.362 T:28931   error <general>: Control 55 in window 10025 has been asked to focus, but it can't
2023-06-03 21:16:13.136 T:29040   error <general>: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                                    - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                                   Error Type: <class 'AttributeError'>
                                                   Error Contents: module 'urllib3.util.ssl_' has no attribute '_ssl_wrap_socket_impl'
                                                   Traceback (most recent call last):
                                                     File "/home/juampi/.kodi/addons/slyguy.disney.plus/default.py", line 3, in <module>
                                                       from resources.lib.plugin import plugin
                                                     File "/home/juampi/.kodi/addons/slyguy.disney.plus/resources/lib/plugin.py", line 3, in <module>
                                                       from slyguy import plugin, gui, userdata, signals, inputstream, settings
                                                     File "/home/juampi/.kodi/addons/script.module.slyguy/resources/modules/slyguy/plugin.py", line 12, in <module>
                                                       from . import router, gui, settings, userdata, inputstream, signals, migrate, bookmarks, mem_cache
                                                     File "/home/juampi/.kodi/addons/script.module.slyguy/resources/modules/slyguy/inputstream.py", line 12, in <module>
                                                       from .session import Session
                                                     File "/home/juampi/.kodi/addons/script.module.slyguy/resources/modules/slyguy/session.py", line 40, in <module>
                                                       orig_ssl_wrap_socket_impl = urllib3.util.ssl_._ssl_wrap_socket_impl
                                                   AttributeError: module 'urllib3.util.ssl_' has no attribute '_ssl_wrap_socket_impl'
                                                   -->End of Python script error report<--
                                                   
2023-06-03 21:16:13.178 T:29039   error <general>: GetDirectory - Error getting plugin://slyguy.disney.plus/
2023-06-03 21:16:13.191 T:28931   error <general>: CGUIMediaWindow::GetDirectory(plugin://slyguy.disney.plus/) failed

```

**Comment by matthuisman** [comment]
i dont need the error - i need the log.
The specific message i need is further back

**Comment by jpcandioti** [comment]
Sorry, I don't understand... This error occurs when i run the disney addon. This is the only thing that shows up in .kodi/temp/kodi.log.

Where should I see it?

**Comment by matthuisman** [comment]
Please provide a full Kodi debug log. That will contain the information I
require

**Comment by matthuisman** [comment]
You will need to enable debug logging in Kodi settings

**Comment by jpcandioti** [comment]
```
2023-06-03 21:31:51.664 T:29703   debug <general>: CGUIMediaWindow::GetDirectory (plugin://slyguy.disney.plus/)
2023-06-03 21:31:51.664 T:29703   debug <general>:   ParentPath = [plugin://slyguy.disney.plus/]
2023-06-03 21:31:51.665 T:29829   debug <general>: Thread waiting start, auto delete: false
2023-06-03 21:31:51.728 T:29781   debug <general>: CAddonDatabase::SetLastUsed[slyguy.disney.plus] took 63 ms
2023-06-03 21:31:51.729 T:29829   debug <general>: CScriptRunner: running add-on script Disney+('plugin://slyguy.disney.plus/', '2', '')
2023-06-03 21:31:51.729 T:29830   debug <general>: Thread LanguageInvoker start, auto delete: false
2023-06-03 21:31:51.729 T:29830   debug <general>: initializing python engine.
2023-06-03 21:31:51.729 T:29830   debug <general>: CPythonInvoker(3, /home/juampi/.kodi/addons/slyguy.disney.plus/default.py): start processing
2023-06-03 21:31:51.765 T:29703   debug <general>: ------ Window Init (DialogBusy.xml) ------
2023-06-03 21:31:51.824 T:29830   debug <general>: -->Python Interpreter Initialized<--
2023-06-03 21:31:51.824 T:29830   debug <general>: 
                                                   
2023-06-03 21:31:51.824 T:29830   debug <general>: CPythonInvoker(3, /home/juampi/.kodi/addons/slyguy.disney.plus/default.py): the source file to load is "/home/juampi/.kodi/addons/slyguy.disney.plus/default.py"
2023-06-03 21:31:51.824 T:29830   debug <general>: CPythonInvoker(3): default python path:
2023-06-03 21:31:51.824 T:29830   debug <general>: CPythonInvoker(3):   /usr/lib/python38.zip
2023-06-03 21:31:51.824 T:29830   debug <general>: CPythonInvoker(3):   /usr/lib/python3.8
2023-06-03 21:31:51.824 T:29830   debug <general>: CPythonInvoker(3):   /usr/lib/python3.8/lib-dynload
2023-06-03 21:31:51.824 T:29830   debug <general>: CPythonInvoker(3):   /home/juampi/.local/lib/python3.8/site-packages
2023-06-03 21:31:51.824 T:29830   debug <general>: CPythonInvoker(3):   /usr/local/lib/python3.8/dist-packages
2023-06-03 21:31:51.824 T:29830   debug <general>: CPythonInvoker(3):   /usr/lib/python3/dist-packages
2023-06-03 21:31:51.824 T:29830   debug <general>: CPythonInvoker(3): adding path:
2023-06-03 21:31:51.824 T:29830   debug <general>: CPythonInvoker(3):  /home/juampi/.kodi/addons/script.module.slyguy/resources/modules
2023-06-03 21:31:51.824 T:29830   debug <general>: CPythonInvoker(3):  /home/juampi/.kodi/addons/slyguy.dependencies/resources/modules
2023-06-03 21:31:51.824 T:29830   debug <general>: CPythonInvoker(3):  /home/juampi/.kodi/addons/slyguy.disney.plus
2023-06-03 21:31:51.824 T:29830   debug <general>: CPythonInvoker(3): adding args:
2023-06-03 21:31:51.824 T:29830   debug <general>: CPythonInvoker(3):  plugin://slyguy.disney.plus/
2023-06-03 21:31:51.824 T:29830   debug <general>: CPythonInvoker(3):  2
2023-06-03 21:31:51.824 T:29830   debug <general>: CPythonInvoker(3):  
2023-06-03 21:31:51.824 T:29830   debug <general>: CPythonInvoker(3):  resume:false
2023-06-03 21:31:51.824 T:29830   debug <general>: CPythonInvoker(3, /home/juampi/.kodi/addons/slyguy.disney.plus/default.py): entering source directory /home/juampi/.kodi/addons/slyguy.disney.plus
2023-06-03 21:31:51.824 T:29830   debug <general>: CPythonInvoker(3, /home/juampi/.kodi/addons/slyguy.disney.plus/default.py): instantiating addon using automatically obtained id of "slyguy.disney.plus" dependent on version 0.0.0 of the xbmc.python api
2023-06-03 21:31:52.343 T:29830   debug <general>: slyguy.disney.plus - OpenSSL 1.1.1f  31 Mar 2020
2023-06-03 21:31:52.344 T:29830   error <general>: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                                    - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                                   Error Type: <class 'AttributeError'>
                                                   Error Contents: module 'urllib3.util.ssl_' has no attribute '_ssl_wrap_socket_impl'
                                                   Traceback (most recent call last):
                                                     File "/home/juampi/.kodi/addons/slyguy.disney.plus/default.py", line 3, in <module>
                                                       from resources.lib.plugin import plugin
                                                     File "/home/juampi/.kodi/addons/slyguy.disney.plus/resources/lib/plugin.py", line 3, in <module>
                                                       from slyguy import plugin, gui, userdata, signals, inputstream, settings
                                                     File "/home/juampi/.kodi/addons/script.module.slyguy/resources/modules/slyguy/plugin.py", line 12, in <module>
                                                       from . import router, gui, settings, userdata, inputstream, signals, migrate, bookmarks, mem_cache
                                                     File "/home/juampi/.kodi/addons/script.module.slyguy/resources/modules/slyguy/inputstream.py", line 12, in <module>
                                                       from .session import Session
                                                     File "/home/juampi/.kodi/addons/script.module.slyguy/resources/modules/slyguy/session.py", line 40, in <module>
                                                       orig_ssl_wrap_socket_impl = urllib3.util.ssl_._ssl_wrap_socket_impl
                                                   AttributeError: module 'urllib3.util.ssl_' has no attribute '_ssl_wrap_socket_impl'
                                                   -->End of Python script error report<--
                                                   
2023-06-03 21:31:52.344 T:29830   debug <general>: onExecutionDone(3, /home/juampi/.kodi/addons/slyguy.disney.plus/default.py)
2023-06-03 21:31:52.348 T:29703   debug <general>: ------ Window Init (DialogNotification.xml) ------
2023-06-03 21:31:52.357 T:29830   debug <general>: .
2023-06-03 21:31:52.382 T:29830   debug <general>: Skipped 1 duplicate messages..
2023-06-03 21:31:52.382 T:29830   debug <general>: Python interpreter stopped
2023-06-03 21:31:52.382 T:29830   debug <general>: Thread LanguageInvoker 139748505024256 terminating
2023-06-03 21:31:52.392 T:29829   error <general>: GetDirectory - Error getting plugin://slyguy.disney.plus/
2023-06-03 21:31:52.392 T:29829   debug <general>: Thread waiting 139748060907264 terminating
2023-06-03 21:31:52.399 T:29703   debug <general>: ------ Window Deinit (DialogBusy.xml) ------
2023-06-03 21:31:52.400 T:29703   error <general>: CGUIMediaWindow::GetDirectory(plugin://slyguy.disney.plus/) failed
2023-06-03 21:31:52.400 T:29703   debug <general>: CGUIMediaWindow::GetDirectory ()
2023-06-03 21:31:52.400 T:29703   debug <general>:   ParentPath = [plugin://slyguy.disney.plus/]
2023-06-03 21:31:52.402 T:29831   debug <general>: Thread waiting start, auto delete: false
2023-06-03 21:31:52.402 T:29831   debug <general>: Thread waiting 139748060907264 terminating
2023-06-03 21:31:52.403 T:29832   debug <general>: Thread BackgroundLoader start, auto delete: false
2023-06-03 21:31:52.405 T:29832   debug <general>: Thread BackgroundLoader 139749676852992 terminating
```

**Comment by matthuisman** [comment]
There are no messages further up about sys.path ?

**Comment by jpcandioti** [comment]
No, there aren't. I was trying it again. Should I need to enable another debug switch?

**Comment by matthuisman** [comment]
just pushed 0.65.7 which adds a bit more logging.
please update slyguy.common to that and then with debug logging enabled - run the addon.

Then from the log, get me the line that has "sys.path" in it

**Comment by matthuisman** [comment]
fixed with slyguy common 0.65.8
